### ISSUE TITLE
Error with logical operators 'OR' and 'UNION'

### ISSUE THREAD

**Issue by o-mouneu** [hgourvest/node-firebird#239]
Hello 
I am using node-firebird v0.9.9 (with npm version 6.14.8) with firebird server v2.1.7.18553, I am encountering the error 'Malformed String' while trying to make query with the logical operators 'OR' and 'UNION'. 

1st example, the following query works well and i get the expected result :
`SELECT P.PAT_ID, P.PAT_PRE AS PRENOM, upper(P.PAT_NOM) AS NOM, upper(P.PAT_NOM)||' '||P.PAT_PRE PAT_NOMPRE, M.mvt_id, MVT_DTENT AS DATE_ENTREE, MVT_DTSOR AS DATE_SORTIE, UPPER(E.ETM_CLE) AS PLATEAU FROM mvt M INNER JOIN UMED ON M.UME_ID=UMED.UME_ID INNER JOIN PAT P ON M.PAT_ID=P.PAT_ID INNER JOIN ETAMVT E ON (M.ETM_ID = E.ETM_ID) WHERE (M.mvt_dtent<='2019/11/01' AND M.mvt_dtsor>='2019/11/01') `

However the following query doesn't work and generate the 'malformed string' error (this the same query but with a supplementary OR condition) :
` SELECT P.PAT_ID, P.PAT_PRE AS PRENOM, upper(P.PAT_NOM) AS NOM, upper(P.PAT_NOM)||' '||P.PAT_PRE PAT_NOMPRE, M.mvt_id, MVT_DTENT AS DATE_ENTREE, MVT_DTSOR AS DATE_SORTIE, UPPER(E.ETM_CLE) AS PLATEAU FROM mvt M INNER JOIN UMED ON M.UME_ID=UMED.UME_ID INNER JOIN PAT P ON M.PAT_ID=P.PAT_ID INNER JOIN ETAMVT E ON (M.ETM_ID = E.ETM_ID) WHERE (M.mvt_dtent<='2019/11/01' AND M.mvt_dtsor>='2019/11/01') OR (M.mvt_dtent>='2019/11/01' AND M.mvt_dtent<='2020/11/01') `

I test those 2 queries with iSQL tool and they both work. I try to make the same query (as the 2st example) under parameterized form but the result it's the same. Finally instead of using 'OR' I try use 'UNION' but the error remains.
Query with UNION instead of OR
`SELECT P.PAT_ID, P.PAT_PRE AS PRENOM, upper(P.PAT_NOM) AS NOM, upper(P.PAT_NOM)||' '||P.PAT_PRE PAT_NOMPRE, M.mvt_id, MVT_DTENT AS DATE_ENTREE, MVT_DTSOR AS DATE_SORTIE, UPPER(E.ETM_CLE) AS PLATEAU FROM mvt M INNER JOIN UMED ON M.UME_ID=UMED.UME_ID INNER JOIN PAT P ON M.PAT_ID=P.PAT_ID INNER JOIN ETAMVT E ON (M.ETM_ID = E.ETM_ID) WHERE (M.mvt_dtent<='2019/11/01' AND M.mvt_dtsor>='2019/11/01')
UNION SELECT P.PAT_ID, P.PAT_PRE AS PRENOM, upper(P.PAT_NOM) AS NOM, upper(P.PAT_NOM)||' '||P.PAT_PRE PAT_NOMPRE, M.mvt_id, MVT_DTENT AS DATE_ENTREE, MVT_DTSOR AS DATE_SORTIE, UPPER(E.ETM_CLE) AS PLATEAU FROM mvt M INNER JOIN UMED ON M.UME_ID=UMED.UME_ID INNER JOIN PAT P ON M.PAT_ID=P.PAT_ID INNER JOIN ETAMVT E ON (M.ETM_ID = E.ETM_ID) WHERE (M.mvt_dtent>='2019/11/01' AND M.mvt_dtent<='2020/11/01') `

I have not escape line in my query when it's executed. There is my code for executing queries : 
`async function runQuery(req, parameters, callback) {
  var options = {};
  options.host = 'CAP-PS-...';
  options.port = 3050;
  options.database = '...';
  options.user = '...';
  options.password = '...';
  options.lowercase_keys = false; 
  options.role = null;           
  options.pageSize = 4096;     

  Firebird.attach(options, function(err, db) {
      if (err) return callback(err);
      db.query(req, parameters, function(err, result) {
        if(err) { return callback(err);
        } else { callback(null, result); }
       db.detach();
      });
  });
}
module.exports.runQuery = runQuery;`

Thanks in advance !
Olivier

**Comment by o-mouneu** [comment]
I found the cause of my error, it was link to the concatenation : 'upper(P.PAT_NOM)||' '||P.PAT_PRE PAT_NOMPRE' in my queries.

After removing, the error left.
